apps/api/src/app/shared/services/cache/keys.ts

```diff
@@ -0,0 +1,174 @@
+import { GetNotificationsFeedCommand } from '../../../widgets/usecases/get-notifications-feed/get-notifications-feed.command';
+import { GetFeedCountCommand } from '../../../widgets/usecases/get-feed-count/get-feed-count.command';
+
+export enum CacheKeyPrefixEnum {
+  MESSAGE_COUNT = 'message_count',
+  FEED = 'feed',
+  SUBSCRIBER = 'subscriber',
+  NOTIFICATION_TEMPLATE = 'notification_template',
+  USER = 'user',
+  INTEGRATION = 'integration',
+  ENVIRONMENT_BY_API_KEY = 'environment_by_api_key',
+}
+
+export enum CacheKeyTypeEnum {
+  ENTITY = 'entity',
+  QUERY = 'query',
+}
+
+export const QUERY_PREFIX = '#query#';
+
+export class KeyGenerator {
+  public static query = () => {
+    return { feed, messageCount };
+  };
+
+  public static entity = () => {
+    return { subscriber, integration, notificationTemplate, user, environmentByApiKey };
+  };
+}
```

Comment:
This one is a POC I am thinking about what could be the easiest way to interact with the methods, with IDE suggestion of the class methods